x-pack/plugins/infra/common/log_sources/resolved_log_source_configuration.ts

```diff
@@ -65,6 +65,7 @@ const resolveLegacyReference = async (
     timestampField: TIMESTAMP_FIELD,
     tiebreakerField: TIEBREAKER_FIELD,
     messageField: sourceConfiguration.fields.message,
+    // @ts-ignore
```

Comment:
Nit: Would be nice with a comment on why this ts-ignore is here so we know when it's safe to remove or understand when the compiler marks it as obsolete

Reply:
Previously `fields` was essentially untyped. This function is typed to return a `DataView.fields` object which differs from the `FieldSpec[]` that is actually returned. Figuring out why `DataView.fields` is expected goes at least a couple of layers deep. I'm not sure I've added anything of value in this rambling.

Reply:
Ah I see, I can make an issue for my team to follow up on. 
So the correct type to use is `FieldSpec[]`?